free-courses-en.md

```diff
@@ -292,7 +292,7 @@
 * [Princeton Algorithms, Part 1](https://www.coursera.org/course/algs4partI)
 * [Problem Solving With Java](https://www.udacity.com/course/intro-to-java-programming--cs046) (Udacity)
 * [Spring 5 Core - An Ultimate Guide](https://www.udemy.com/learn-spring-5-core-from-scratch/) - Somnath Musib (Udemy)
-
+* [Java Design Patterns and Architecture](https://www.udemy.com/course/java-design-patterns-tutorial/)-  John Purcell (Udemy)
 
 ### JavaScript
```

Comment:
Hello, the Travis CI build fails for 3 reasons:

1. It is not in the correct position (needs to be alphabetical) So in this case "Java Design ..." should come first!
2. There needs to be a space between the https link and the "-". I show this in the suggested code below.
3. Lastly, the end of this section has 1 blank line before JavaScript, but there should only be 2! 

Refer to [CONTRIBUTING](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md) for the above reasons.

Fixing these will make the Travis CI build work. 

```suggestion
* [Java Design Patterns and Architecture](https://www.udemy.com/course/java-design-patterns-tutorial/) -  John Purcell (Udemy)


### JavaScript
```

Reply:
fixed !! but PR is failing

Reply:
Hello, it is due to #4606 mainly